Issues 37376 and 35065: Pivot tables not respecting column displayWidths

labkey-martyp opened the pull request:
#### Rationale
In pivot tables, neither the pivot columns nor the non-pivot columns are respecting display widths set on the column metadata, such as in .query.xml files.  This PR fixes both cases.  Since the table-layout is default to auto, the tables are essentially resizing to contents and the container size.  This PR moves the metadata defined styles from the th tags to a div inside the th tag around the content.  The tables will then resize to the content.  Also removed the width: 100% from pivot queries only, as this was stretching out those column widths in some cases and ignoring the set widths.  Tested on pivot and non-pivot tables, with and without data in Chrome, FF and Edge.

#### Related Pull Requests
https://github.com/LabKey/testAutomation/pull/427

#### Changes
- Move style applied to table column headers to a div wrapping the contents in the th tag instead of on the th itself. For consistency, the div wrapper is always on the th content regardless if style is applied.
- For pivot queries only, set width: auto on upper two rows of th tags

labkey-martyp committed:
Add class for pivot width set to inherit and apply to pivot tables

labkey-martyp committed:
Merge branch 'develop' into fb_37376_pivot_table_width

labkey-martyp committed:
Revert "Add class for pivot width set to inherit and apply to pivot tables"

This reverts commit 66e0bf7c67482ec8c090810559f25ad2e94286c8.

labkey-martyp committed:
Apply column header styles to inner div

- Allows table-layout which is auto by default to adjust to contents
- Allows setting display width in pivot queries on both pivot and non-pivot columns

labkey-martyp committed:
Pivot queries trying to fill width ignore specified width on column

labkey-martyp committed:
Merge branch 'develop' into fb_37376_pivot_table_width

labkey-jeckels commented on api/src/org/labkey/api/data/DisplayColumn.java:
```diff
@@ -734,6 +719,21 @@ public void renderGridHeaderCell(RenderContext ctx, Writer out, String headerCla
+        {
+            out.write("<div style=\"");
+            out.write(style);
+            out.write("\">");
```
Is this right? Seems like we wouldn't want to close it here, and you're also closing it at line 759

labkey-martyp replied:
That was just closing the opening tag.  But I've updated this a bit to always include the div tag around th content, whether there is style applied or not.  This is for a consistent DOM across all column headers, for CSS and testing.

labkey-martyp committed:
Update to always have div on header contents for consistency

- CSS consistency
- Test consistency

labkey-martyp committed:
Merge branch 'develop' into fb_37376_pivot_table_width

labkey-martyp committed:
Update test

- Selenium getText() function returning new line for col headers on embedded tables with this change

labkey-jeckels reviewed (approved):
I was misreading the code on the <div> generation so my previous comment was wrong. Having the consistent structure like you do now makes sense too.
